Assets/HoloToolkit-Examples/UX/Scripts/ProgressExampleLaunchButton.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity.Buttons;
+using HoloToolkit.UX.Progress;
+using UnityEngine;
+
+namespace HoloToolkit.Examples.UX
+{
+    public class ProgressExampleLaunchButton : MonoBehaviour
+    {
+        [Header("Which Indicator style is desired?")]
+        [SerializeField]
+        private ProgressIndicator.IndicatorStyleEnum indicatorStyle;
+
+        [Header("Which Progress style is desired?")]
+        [SerializeField]
+        private ProgressIndicator.ProgressStyleEnum progressStyle;
+
+        public ProgressIndicator.IndicatorStyleEnum IndicatorStyle
+        {
+            get
+            {
+                return indicatorStyle;
+            }
+
+            set
+            {
+                indicatorStyle = value;
+            }
+        }
+
+        public ProgressIndicator.ProgressStyleEnum ProgressStyle
+        {
+            get
+            {
+                return progressStyle;
+            }
+
+            set
+            {
+                progressStyle = value;
+            }
+        }
+
+        private void OnEnable()
+        {
+            GetComponent<Button>().OnButtonClicked += OnButtonClicked;
```

Comment:
let's cache the button in awake or start and register it then.  (we don't have to worry about if it's disabled cause the click will never happen.